tests/writer_test.py

```diff
@@ -0,0 +1,471 @@
+        actual = test_helpers.write_to_lines(glyphs_object)
+        # print(expected)
+        # print(actual)
+        self.assertLinesEqual(
+            expected, actual,
+            "The writer has not produced the expected output")
+
+    def assertWritesValue(self, glyphs_value, text):
+        """Assert that the writer produces the given text for the given value."""
+        expected = dedent("""\
+        {{
+        writtenValue = {0};
+        }}
+        """).format(text).splitlines()
+        # We wrap the value in a dict to use the same test helper
+        actual = test_helpers.write_to_lines({'writtenValue': glyphs_value})
+        self.assertLinesEqual(
+            expected, actual,
+            "The writer has not produced the expected output")
+
+    def test_write_font_attributes(self):
+        """Test the writer on all GSFont attributes"""
+        font = classes.GSFont()
+        # List of properties from https://docu.glyphsapp.com/#gsfont
+        # parent: not handled because it's internal and read-only
+        # masters
+        m1 = classes.GSFontMaster()
+        m1.id = "M1"
+        font.masters.insert(0, m1)
+        m2 = classes.GSFontMaster()
+        m2.id = "M2"
+        font.masters.insert(1, m2)
+        # instances
+        i1 = classes.GSInstance()
+        i1.name = "MuchBold"
+        font.instances.append(i1)
+        # glyphs
+        g1 = classes.GSGlyph()
+        g1.id = 'G1'
+        font.glyphs.append(g1)
+        # classes
+        c1 = classes.GSClass()
+        c1.name = "C1"
+        font.classes.append(c1)
+        # features
+        f1 = classes.GSFeature()
+        f1.name = "F1"
+        font.features.append(f1)
+        # featurePrefixes
+        fp1 = classes.GSFeaturePrefix()
+        fp1 = "FP1"
+        font.featurePrefixes.append(fp1)
+        # copyright
+        font.copyright = "Copyright Bob"
+        # designer
+        font.designer = "Bob"
+        # designerURL
+        font.designerURL = "bob.me"
+        # manufacturer
+        font.manufacturer = "Manu"
+        # manufacturerURL
+        font.manufacturerURL = "manu.com"
+        # versionMajor
+        font.versionMajor = 2
+        # versionMinor
+        font.versionMinor = 104
+        # date
+        font.date = glyphs_datetime('2017-10-03 07:35:46 +0000')
+        # familyName
+        font.familyName = "Sans Rien"
+        # upm
+        font.upm = 2000
+        # note
+        font.note = "Was bored, made this"
+        # kerning
+        font.kerning = OrderedDict([
+            ('M1', OrderedDict([
+                ('@MMK_L_G1', OrderedDict([
+                    ('@MMK_R_G1', 0.1)
+                ]))
+            ]))
+        ])
+        # userData
+        font.userData = {
+            'a': 'test',
+            'b': [1, {'c': 2}]
+        }
+        # grid
+        font.grid = 35
+        # gridSubDivisions
+        # FIXME: (jany) In this library it is called "gridSubDivision" (no s)
```

Comment:
It’s "gridSubDivisions" in [GlyphsSDK](https://github.com/schriftgestalt/GlyphsSDK/blob/master/ObjectWrapper/GlyphsApp/__init__.py#L2530). We can fix the typo in classes.py.